src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisCard.tsx

```diff
@@ -0,0 +1,724 @@
+      setReportTime,
+    ]
+  );
+
+  const generateReport = React.useCallback(() => {
+    addSubscription(
+      context.target.target().subscribe((target) => {
+        handleLoading();
+        setTargetConnectURL(target.connectUrl);
+        dispatch(automatedAnalysisAddTargetIntent(target.connectUrl));
+        // query for designated automated-analysis profiling recording
+        addSubscription(
+          queryActiveRecordings(target.connectUrl)
+            .pipe(
+              first(),
+              map((v) => v.data.targetNodes[0].recordings.active.data[0] as Recording),
+              tap((recording) => {
+                if (recording == null) {
+                  throw new Error(NO_RECORDINGS_MESSAGE);
+                }
+              })
+            )
+            .subscribe({
+              next: (recording) => {
+                context.reports
+                  .reportJson(recording, target.connectUrl)
+                  .pipe(first())
+                  .subscribe({
+                    next: (report) => {
+                      categorizeEvaluation(report);
+                    },
+                    error: (_) => {
+                      handleStateErrors(FAILED_REPORT_MESSAGE);
+                    },
+                  });
+              },
+              error: (_) => {
+                // try generating report on cached or archived recordings
+                handleEmptyRecordings(target.connectUrl);
+              },
+            })
+        );
+      })
+    );
+  }, [
+    addSubscription,
+    dispatch,
+    context.api,
+    context.target,
+    context.reports,
+    automatedAnalysisAddTargetIntent,
+    setTargetConnectURL,
+    categorizeEvaluation,
+    queryActiveRecordings,
+    handleEmptyRecordings,
+    handleLoading,
+    handleStateErrors,
+  ]);
+
+  const startProfilingRecording = React.useCallback(() => {
+    const config = context.settings.automatedAnalysisRecordingConfig();
+    const attributes = automatedAnalysisConfigToRecordingAttributes(config);
+    addSubscription(
+      context.api.createRecording(attributes).subscribe((resp) => {
+        if (resp && (resp.ok || resp.status === 400)) {
+          // in-case the recording already exists
+          generateReport();
+        } else {
+          handleStateErrors(RECORDING_FAILURE_MESSAGE);
+        }
+      })
+    );
+  }, [
+    addSubscription,
+    context.api,
+    context.settings,
+    context.settings.automatedAnalysisRecordingConfig,
+    generateReport,
+    handleStateErrors,
+  ]);
+
+  const handleErrorView = React.useCallback((): [string, undefined | (() => void)] => {
+    if (errorMessage === NO_RECORDINGS_MESSAGE) {
+      return ['', undefined];
+    } else if (errorMessage === RECORDING_FAILURE_MESSAGE) {
+      return ['Retry starting recording', startProfilingRecording];
+    } else if (errorMessage === FAILED_REPORT_MESSAGE) {
+      return ['Retry loading report', generateReport];
+    } else {
```

Comment:
Any handler for auth retry?

Reply:
What happens right now, is if I use a authRetry callback handler like 
```
  const authRetry = React.useCallback(() => {
    context.target.setAuthFailure();
  }, [context.target]);
```
Nothing will pop up if you press the Retry text because there is actually nothing to render that would cause a JMX auth error. Because GraphQL does not trigger any auth failure. So I am thinking once the dry run request gets merged, I might be able to use that to test a connection and retry the auth that way? 

For now, you it will call generateReport again, and then to create a recording, this would require auth and then the modal will pop up again.

Reply:
I think the function in authRetry should be:

```typescript
const authRetry = React.useCallback(() => {
  context.target.setAuthRetry();
}, [context.target]);
```

Could u try again?

Reply:
I've tried it. It doesn't work because there is nothing that will automatically render that calls an api call that requires JMX authentication so the button won't do anything. It is only afterwards, when you press "Create recording" will the auth modal popup.